src/engine.rs

```diff
@@ -663,6 +485,29 @@ impl FileEngine {
             inner: Arc::new(engine),
         }
     }
+
+    // Only gc but not purge.
+    fn do_gc(&self, raft_group_id: u64, to: u64) -> usize {
+        self.inner.compact_to(raft_group_id, to) as usize
+    }
+
+    // Do purge and update some internal stats.
+    fn do_purge(&self) {
+        // TODO: rewrite inactive logs.
+        self.inner.purge_expired_files().unwrap();
+        self.inner.cache_stats.on_purge();
```

Comment:
I think the calculating logic of on_purge is not correct before the rewrite inactive logs logic is done.